uikit/views/src/main/java/com/sygic/maps/uikit/views/navigation/infobar/Infobar.kt

```diff
@@ -0,0 +1,286 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sygic.maps.uikit.views.navigation.infobar
+
+import android.animation.LayoutTransition
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.ImageButton
+import android.widget.TextView
+import androidx.annotation.CallSuper
+import androidx.annotation.ColorRes
+import androidx.annotation.DrawableRes
+import androidx.constraintlayout.widget.ConstraintLayout
+import com.sygic.maps.uikit.views.R
+import com.sygic.maps.uikit.views.common.extensions.EMPTY_STRING
+import com.sygic.maps.uikit.views.common.extensions.backgroundTint
+import com.sygic.maps.uikit.views.common.extensions.tint
+import com.sygic.maps.uikit.views.databinding.LayoutInfobarInternalBinding
+import com.sygic.maps.uikit.views.navigation.infobar.items.InfobarItemsHolder
+
+/**
+ * A [Infobar] view can be used as an visual presentation component for the navigation info data (eta, distanceToEnd,
+ * altitude etc.) and as user interaction component. It contains two [ImageButton]'s (left/right) and pre-customized
+ * primary/secondary [TextView] which can be controlled with the [InfobarItemsHolder] class.
+ *
+ * The [Infobar] design can be completely changed with the custom _infoBarStyle_ or the standard android
+ * attributes as _background_, _navigationTextColorPrimary_ or _navigationTextColorSecondary_ can be used. The [Infobar]
+ * view has also own attributes such as _leftButtonIcon_, _leftButtonIconTint_, _leftButtonBackground_, _leftButtonBackgroundTint_,
+ * _rightButtonIcon_, _rightButtonIconTint_, _rightButtonBackground_ and _rightButtonBackgroundTint_.
+ */
+@Suppress("unused", "MemberVisibilityCanBePrivate")
+open class Infobar @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = R.attr.infoBarStyle,
+    defStyleRes: Int = R.style.SygicInfobarStyle
+) : ConstraintLayout(context, attrs, defStyleAttr) {
+
+    private val binding: LayoutInfobarInternalBinding =
+        LayoutInfobarInternalBinding.inflate(LayoutInflater.from(context), this, true)
+
+    private var layoutMargin: Int = 0
+    private var layoutMarginTop: Int = 0
+    private var layoutMarginBottom: Int = 0
+    private var layoutMarginStart: Int = 0
+    private var layoutMarginEnd: Int = 0
+
+    init {
+        isClickable = true
+        layoutTransition = LayoutTransition()
+
+        attrs?.let { attributeSet ->
+            @Suppress("Recycle")
```

Comment:
vyzera ze za "chvilu" uz to nebude potrebne https://issuetracker.google.com/issues/139566120